addOns/zest/src/main/java/org/zaproxy/zap/extension/zest/ExtensionZest.java

```diff
@@ -545,9 +554,18 @@ public ScriptNode add(ZestScriptWrapper script, boolean display) {
     }
 
     public void display(ZestScriptWrapper script, ScriptNode node, boolean expand) {
+        display(script, node, expand, true);
+    }
+
+    public void display(
+            ZestScriptWrapper script, ScriptNode node, boolean expand, boolean allowFocus) {
         if (View.isInitialised() && this.getExtScript().getScriptUI() != null) {
-            this.getExtScript().getScriptUI().selectNode(node, expand);
-            this.getExtScript().getScriptUI().displayScript(script);
+            if (allowFocus) {
+                // This function internally calls set script, which is not trivial to change at the
+                // moment
+                this.getExtScript().getScriptUI().selectNode(node, expand);
+            }
```

Comment:
Had to do it this way because this is not trivial to change. This function calls set script function which sets the focus. 
Here is the stack trace for your reference: 
```
java.lang.Exception
	at org.zaproxy.zap.extension.scripts.ConsolePanel.setScript(ConsolePanel.java:479)
	at org.zaproxy.zap.extension.scripts.ExtensionScriptsUI.displayScript(ExtensionScriptsUI.java:423)
	at org.zaproxy.zap.extension.scripts.ExtensionScriptsUI.displayScript(ExtensionScriptsUI.java:392)
	at org.zaproxy.zap.extension.scripts.ScriptsListPanel.selectionChanged(ScriptsListPanel.java:657)
	at org.zaproxy.zap.extension.scripts.ScriptsListPanel.lambda$getTree$4(ScriptsListPanel.java:640)
	at java.desktop/javax.swing.JTree.fireValueChanged(JTree.java:2967)
	at java.desktop/javax.swing.JTree$TreeSelectionRedirector.valueChanged(JTree.java:3456)
	at java.desktop/javax.swing.tree.DefaultTreeSelectionModel.fireValueChanged(DefaultTreeSelectionModel.java:641)
	at java.desktop/javax.swing.tree.DefaultTreeSelectionModel.notifyPathChange(DefaultTreeSelectionModel.java:1111)
	at java.desktop/javax.swing.tree.DefaultTreeSelectionModel.setSelectionPaths(DefaultTreeSelectionModel.java:297)
	at java.desktop/javax.swing.tree.DefaultTreeSelectionModel.setSelectionPath(DefaultTreeSelectionModel.java:191)
	at java.desktop/javax.swing.JTree.setSelectionPath(JTree.java:1656)
	at org.zaproxy.zap.extension.scripts.ScriptsListPanel.showInTree(ScriptsListPanel.java:714)
	at org.zaproxy.zap.extension.scripts.ExtensionScriptsUI.selectNode(ExtensionScriptsUI.java:802)
	at org.zaproxy.zap.extension.zest.ExtensionZest.display(ExtensionZest.java:568)
	at org.zaproxy.zap.extension.zest.ExtensionZest.add(ExtensionZest.java:550)
	at org.zaproxy.zap.extension.zest.dialogs.ZestRecordScriptDialog.save(ZestRecordScriptDialog.java:277)
	at org.zaproxy.zap.view.StandardFieldsDialog.savePressed(StandardFieldsDialog.java:428)
	at org.zaproxy.zap.view.StandardFieldsDialog$3.actionPerformed(StandardFieldsDialog.java:411)
	at java.desktop/javax.swing.AbstractButton.fireActionPerformed(AbstractButton.java:1967)
	at java.desktop/javax.swing.AbstractButton$Handler.actionPerformed(AbstractButton.java:2308)
	at java.desktop/javax.swing.DefaultButtonModel.fireActionPerformed(DefaultButtonModel.java:405)
	at java.desktop/javax.swing.DefaultButtonModel.setPressed(DefaultButtonModel.java:262)
	at java.desktop/javax.swing.plaf.basic.BasicButtonListener.mouseReleased(BasicButtonListener.java:279)
	at java.desktop/java.awt.Component.processMouseEvent(Component.java:6635)
	at java.desktop/javax.swing.JComponent.processMouseEvent(JComponent.java:3342)
	at java.desktop/java.awt.Component.processEvent(Component.java:6400)
	at java.desktop/java.awt.Container.processEvent(Container.java:2263)
	at java.desktop/java.awt.Component.dispatchEventImpl(Component.java:5011)
	at java.desktop/java.awt.Container.dispatchEventImpl(Container.java:2321)
```